src/modules/ens/sagas.ts

```diff
@@ -258,6 +258,10 @@ function* handleSetENSContentRequest(action: SetENSContentRequestAction) {
     )
 
     yield put(setENSContentSuccess(ens, content, land, from.toString(), wallet.chainId, txHash))
+
+    if (!land) {
+      yield put(replace(locations.activity()))
```

Comment:
Redirect to activity only when unlinking the name from the land. Only on this case because when linking a name, the flow continues on the same page and would be affected by the redirect

Reply:
Could we move this to the `modules/location/saga.ts` saga? So we don't create a dependency on the location module within this one. I noticed that there are already some imports from the ens module to the modal and location modules, but those can be fixed on another PR. What I mean by moving it to the location saga is taking the `SET_ENS_CONTENT_SUCCESS` from there and doing the redirect on that saga, like this:

```ts
export function* locationSaga() {
  yield all([
    /* other sagas... */
    takeLatest(SET_ENS_CONTENT_SUCCESS, handleSetENSContentSuccess),
  ])
}

function* handleSetENSContentSuccess(action: SetENSContnetSuccessAction) {
  const { land } = action.payload
  if (!land) {
    yield put(replace(locations.activity()))
  }
}
```

Reply:
Good idea.
In order to completely remove the locations module I had to do the same thing for the claim name sagas which did the same.

```ts
function* handleClaimNameRequest(action: ClaimNameRequestAction) {
  const { name } = action.payload
  try {
    const [wallet, eth]: [Wallet, Eth] = yield getWallet()
    const from = Address.fromString(wallet.address)

    const controllerContract = new DCLController(eth, Address.fromString(CONTROLLER_ADDRESS))
    const tx: SendTx<TransactionReceipt> = yield call(() => controllerContract.methods.register(name, from).send({ from }))
    const txHash: string = yield call(() => tx.getTxHash())

    const ens: ENS = {
      address: wallet.address,
      name: name,
      subdomain: getDomainFromName(name),
      resolver: Address.ZERO.toString(),
      content: Address.ZERO.toString()
    }
    yield put(claimNameSuccess(ens, name, wallet.address, wallet.chainId, txHash))
    yield put(closeModal('ClaimNameFatFingerModal'))
    yield put(push(locations.activity())) -> HERE
  } catch (error) {
    const ensError: ENSError = { message: error.message }
    yield put(claimNameFailure(ensError))
  }
}
```

Reply:
Maybe it would also be nice to do the same for the closeModal dispatch. I've seen that there is a `modal/sagas.ts` that can be used